Using junit5 with auto detect

vrdhn opened the pull request.

vrdhn committed:
Using junit5 console launcher for automatic test detection

bremac commented:
Using the ConsoleLauncher looks good to me, but we need to combine `JAVA_TEST_CLASSES` and `JAVA_ALL_TEST_CLASSES` into the test JAR so that the launcher discovers them. See https://github.com/sightmachine/rocksdb/pull/4/files#diff-31ed790e942fadafe334a13765d2379b7b89b1c3d65a3973a94d55a77624c5db

bremac commented on .circleci/config.yml:
```diff
@@ -158,6 +158,8 @@ jobs:
             mkdir test-jars/
             cp java/test-libs/*.jar test-jars/
             cp build/java/rocksdbjni_test_classes.jar test-jars/
+            rm test-jars/*junit-4*
```
I think we could just remove the JUnit 4 JAR from the CMakeLists.txt?

vrdhn replied:
well, these guys have their own s3 bucket from where they download java deps in the CMake.
The Right way is obviously to put junit5 in there, ... so i just chose the [worst](https://www.jwz.org/doc/worse-is-better.html) way (-:

bremac reviewed (approved):
LGTM. I'm going to get all of the test classes into the test JAR and then merge this. :+1:

bremac committed:
Put all tests classes in JAVA_TEST_CLASSES

This ensures that all of the test classes are present so that they can
be discovered by the JUnit5 runner.
